lib/msf/core/payload_generator.rb

```diff
@@ -195,6 +199,24 @@ def add_shellcode(shellcode)
       end
     end
 
+    # This method takes the shellcode generated so far and adds shellcode from
+    # a supplied file to the beginning of the shellcode.
+    # @param shellcode [String] The shellcode to be prepended
+    # @return [String] the combined shellcode
+    def prepend_shellcode(shellcode)
+      print_line("Prepending code")
+      if prepend_code.present?
+        cli_print "Adding shellcode from #{prepend_code} to the beginning of the payload"
+        shellcode_file = File.open(prepend_code)
+        shellcode_file.binmode
+        added_code = shellcode_file.read
+        shellcode_file.close
+        added_code << shellcode
+      else
+        shellcode.dup
+      end
```

Comment:
```suggestion
      return shellcode unless prepend_code.present?
      
      cli_print "Adding shellcode from #{prepend_code} to the beginning of the payload"
      File.binread(prepend_code) + shellcode
```

Reply:
@timwr no; we have the `add-shellcode` @adfoster-r7 shows that launches a second thread (but only for win32).  This was made just to run some prep code before the payload.

Reply:
@adfoster-r7 yup; I was trying to remember the exact method for binread, but I saw that the code block did exactly what I wanted, and this was just a discussion PR, so 🤷 .